game/i18n.hh

```diff
@@ -2,26 +2,130 @@
 
 #include <boost/locale.hpp>
 #include <iostream>
+#include <string>
+#include <map>
+
+#include "configuration.hh"
 #include "fs.hh"
 
 #define _(x) boost::locale::translate(x).str()
 #define translate_noop(x) x
 
 class TranslationEngine {
 public:
-	TranslationEngine(const char *package) {
-		boost::locale::generator gen;
-		gen.add_messages_path(getLocaleDir().string());
-		gen.add_messages_domain(package);
-		try {
-			std::locale::global(gen(boost::locale::util::get_system_locale(true)));
-		} catch (...) {
-			std::clog << "locale/warning: Unable to detect locale, will try to fallback to en_US.UTF-8" << std::endl;
-			std::locale::global(gen("en_US.UTF-8"));
-		}
+	TranslationEngine(const char *package) : m_package(package) {
+
+		initializeAllLanguages();
+		populateLanguages(GetAllLanguages());
+
+		std::clog << "locale/debug: Checking if language is set in config." << std::endl;
+
+		auto prefValue = config["game/language"].getEnumName();
```

Comment:
why not storing english tong name instead of enums?

Reply:
Hmmm @Lord-Kamina suggested it this way like he did with Audio backends